bindings/pyroot_experimental/PyROOT/src/CPPInstancePyz.cxx

```diff
@@ -0,0 +1,123 @@
+    if (!PyArg_ParseTuple(args, const_cast< char* >( "O!O!:__expand__" ), &PyBytes_Type, &pybuf, &PyBytes_Type, &pyname))
+        return 0;
+    const char* clname = PyBytes_AS_STRING(pyname);
+    // TBuffer and its derived classes can't write themselves, but can be created
+    // directly from the buffer, so handle them in a special case
+    void* newObj = 0;
+    if (strcmp(clname, "TBufferFile" ) == 0) {
+        TBufferFile* buf = new TBufferFile( TBuffer::kWrite );
+        buf->WriteFastArray(PyBytes_AS_STRING(pybuf), PyBytes_GET_SIZE(pybuf));
+        newObj = buf;
+    } else {
+        // use the PyString macro's to by-pass error checking; do not adopt the buffer,
+        // as the local TBufferFile can go out of scope (there is no copying)
+        TBufferFile buf(TBuffer::kRead, PyBytes_GET_SIZE(pybuf), PyBytes_AS_STRING(pybuf), kFALSE );
+        newObj = buf.ReadObjectAny(0);
+    }
+    PyObject* result = BindCppObject(newObj, Cppyy::GetScope(clname));
+    if (result) {
+        // this object is to be owned by the interpreter, assuming that the call
+        // originated from there
+        ((CPPInstance*)result)->PythonOwns();
+    }
+    return result;
+}
+
+/// PyROOT object proxy pickle support
+PyObject* op_reduce(CPPInstance* self) {
+    // Turn the object proxy instance into a character stream and return for
+    // pickle, together with the callable object that can restore the stream
+    // into the object proxy instance.
+
+    // keep a borrowed reference around to the callable function for expanding;
+    // because it is borrowed, it means that there can be no pickling during the
+    // shutdown of the libPyROOT module
+    static PyObject* s_expand = PyDict_GetItemString(PyModule_GetDict(PyROOT::gRootModule), const_cast<char*>("_CPPInstance__expand__"));
+
+    // TBuffer and its derived classes can't write themselves, but can be created
+    // directly from the buffer, so handle them in a special case
+    static Cppyy::TCppType_t s_bfClass = Cppyy::GetScope("TBufferFile");
+    TBufferFile* buff = 0;
+    if (s_bfClass == self->ObjectIsA()) {
+        buff = (TBufferFile*)self->GetObject();
+    } else {
+        // no cast is needed, but WriteObject taking a TClass argument is protected,
+        // so use WriteObjectAny()
+        static TBufferFile s_buff(TBuffer::kWrite);
+        s_buff.Reset();
+        // to delete
+        if (s_buff.WriteObjectAny(self->GetObject(), TClass::GetClass(Cppyy::GetScopedFinalName(self->ObjectIsA()).c_str())) != 1) {
+            PyErr_Format(PyExc_IOError, "could not stream object of type %s", Cppyy::GetScopedFinalName(self->ObjectIsA()).c_str());
+            return 0;
+        }
+        buff = &s_buff;
+    }
+    // use a string for the serialized result, as a python buffer will not copy
+    // the buffer contents; use a string for the class name, used when casting
+    // on reading back in (see RootModule.cxx:TObjectExpand)
```

Comment:
This reference to `RootModule.cxx:TObjectExpand` is no longer valid. Right now the equivalent function is the `CPPInstanceExpand` define above.